File: HelpSoftware/Help/EF/ArticleNomination.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Help
{
  public class ArticleNomination
  {
    [Key]
    public int ID_ArticleNomination { get; set; }

    public User NominatedUser { get; set; } 

    public User RequestedUser { get; set; }

    public string ArticleTitle { get; set; }

    public string ArticleDescription { get; set; }

    public string Comment { get; set; }

    public ICollection<Topic> Topics { get; set; }

    public ICollection<Tag> Tags { get; set; }

    public DateTime NominationedDate { get; set; }

    public bool IsSeen { get; set; }

    public DateTime SeenDate { get; set; }
  }
}

File: HelpSoftware/Help/Help.Web/Web/src/app/EF/ArticleNomination.ts
import { User, Topic, Tag } from "./EF";

export class ArticleNomination {
  ID_ArticleNomination: Number;
  NominatedUser: User;
  RequestedUser: User;
  ArticleTitle: string;
  ArticleDescription: string;
  Comment: string;
  Topics: Topic[];
  Tags: Tag[];
  NominationedDate: Date;
  IsSeen: boolean;
  SeenDate: Date;

  deserialize(input) {
    this.ID_ArticleNomination = input.ID_ArticleNomination;
    this.NominatedUser = new User().deserialize(input.NominatedUser);
    this.RequestedUser = new User().deserialize(input.RequestedUser);
    this.ArticleTitle = input.ArticleTitle;
    this.ArticleDescription = input.ArticleDescription;
    this.Comment = input.Comment;
    this.NominationedDate = input.NominationedDate;
    this.IsSeen = input.IsSeen;
    this.SeenDate = input.Date;
    return this;
  }
}
